frontend/admin/resources/js/components/skill/UpdateSkill.tsx

```diff
@@ -0,0 +1,271 @@
+import React from "react";
+import { useIntl } from "react-intl";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import pick from "lodash/pick";
+import sortBy from "lodash/sortBy";
+import { toast } from "react-toastify";
+import { Submit, Input, MultiSelect, TextArea } from "@common/components/form";
+import { notEmpty } from "@common/helpers/util";
+import { navigate } from "@common/helpers/router";
+import { getLocale } from "@common/helpers/localize";
+import { unpackIds } from "@common/helpers/formUtils";
+
+import { errorMessages, commonMessages } from "@common/messages";
+import { useAdminRoutes } from "../../adminRoutes";
+import {
+  Skill,
+  SkillFamily,
+  UpdateSkillInput,
+  UpdateSkillMutation,
+  useUpdateSkillMutation,
+  useGetUpdateSkillDataQuery,
+} from "../../api/generated";
+import DashboardContentContainer from "../DashboardContentContainer";
+
+type Option<V> = { value: V; label: string };
+
+type FormValues = Pick<Skill, "name" | "description"> & {
+  keywords: string;
+  families: string[];
+};
+
+interface UpdateSkillFormProps {
+  initialSkill: Skill;
+  families: SkillFamily[];
+  handleUpdateSkill: (
+    id: string,
+    data: UpdateSkillInput,
+  ) => Promise<UpdateSkillMutation["updateSkill"]>;
+}
+
+export const UpdateSkillForm: React.FunctionComponent<UpdateSkillFormProps> = ({
+  initialSkill,
+  families,
+  handleUpdateSkill,
+}) => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+  const paths = useAdminRoutes();
+  const sortedFamilies = sortBy(families, (family) => {
+    return family.name?.[locale]?.toLocaleUpperCase();
+  });
+
+  const dataToFormValues = (data: Skill): FormValues => ({
+    ...data,
+    keywords: data.keywords?.join(", ") || "",
+    families: unpackIds(data?.families),
+  });
+
+  const formValuesToSubmitData = (values: FormValues): UpdateSkillInput => ({
+    ...values,
+    keywords: values.keywords.split(",").map((key) => key.trim()),
```

Comment:
Lol, this is a bit janky but we never received a wireframe from designers and this is the backend admin so I say let's leave it.  I suggest adding a helper button (like on the key field) to both _create_ and _update_ to explain that the field holds multiple comma-delimited values.

Reply:
Oh, please check for empty strings too.  Like if the field value is just `,` for example.

Reply:
It turns out it does load an empty string. I've fixed it now though.